convert_fendl.py

```diff
@@ -41,90 +42,308 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
+            # header to include in every file
+            header_line = base_file.readline()
+            current_file_str += header_line
+            
+            for line in base_file:
+                
+                current_line_no = int(line.split()[-1])
+                
+                # Start of a new nuclide
+                if current_line_no < last_line_no:
+                    # Create a temporary file with the data 
+                    new_endf = open('tmp', 'w+')
+                    new_endf.write(current_file_str)
+                    new_endf.seek(0)
+                    
+                    # Use the ENDF evaluation method to read the name
+                    ev = openmc.data.endf.Evaluation(new_endf)
+                    new_endf.close()
+                    
+                    z = ev.target['atomic_number']
+                    a = ev.target['mass_number']
+                    new_filename = openmc.data.data.ATOMIC_SYMBOL[z]
+                    new_filename = new_filename + str(a) + ".endf"
 
-# Get a list of all ACE files, excluding files ending with _ which are old incorrect files kept in the release for backwards compatability
-neutron_files = [
-    f
-    for f in release_details[args.release]['neutron_files']
-    if not f.name.endswith('_') and not f.name.endswith('.xsd')
-]
+                    os.rename('tmp', new_filename)
+                    
+                    # Need to add a header line if there isn't one at the start of
+                    # new nuclide
+                    if current_line_no == 1:
+                        current_file_str = header_line
 
-# Create output directory if it doesn't exist
-args.destination.mkdir(parents=True, exist_ok=True)
+                current_file_str += line
+                last_line_no = current_line_no
+        
+        os.chdir(cwd)
+
+# ==============================================================================
+# GENERATE HDF5 LIBRARY
 
 library = openmc.data.DataLibrary()
 
-for filename in sorted(neutron_files):
+warn_k39 = False    # Flag for K-39 error in FENDL-3.0 release
+
+for particle in args.particles:
+    # Create output directories if it doesn't exist
+    particle_destination = args.destination / particle
+    particle_destination.mkdir(parents=True, exist_ok=True)
+
+    particle_details = release_details[args.release][particle]
+
+    if particle == 'neutron':
+        # Get a list of all ACE files, excluding files ending with _ which are 
+        # old incorrect files kept in the release for backwards compatability
+        neutron_files = [
+            f
+            for f in particle_details['ace_files']
+            if not f.name.endswith('_') and not f.name.endswith('.xsd')
+        ]
+        for filename in sorted(neutron_files):
+            # Check for Inf values in K-39 ace file for FENDL-3.0
+            if args.release == '3.0' and filename.name == '19K_039.ace':
```

Comment:
I was wondering if it is possible to move these odd cases into the release_details dictionary. I can't quite figure out how to do it elegantly but if you have any thoughts on how we can fully specify the release (including odd files and how to handle them) in the release_details then this would bring us one step closer to a generic nuclear data processing script

Reply:
No worries if this is too tricky, we do use this approach elsewhere so I am sure it is fine.